pypicloud/cache/dynamo.py

```diff
@@ -22,6 +26,86 @@
 LOG = logging.getLogger(__name__)
 
 
+def _decide_between_versions(contender, current, allow_overwrite):
```

Comment:
I decided to also respect it during cache rebuilds for two reasons:
 * Some people might have setups where packages might just appear on S3
 * It confers some more determinism to solving the ordering conundrum.

Thinking about it again, since it doesn't provide full determinism (files a/pkg and b/pkg, if created at the same time, are still not orderable between them) and it introduced a conceptual dependency that's surprising for you, I'll just remove it and make the earliest win.